OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -73,31 +80,15 @@ public static Target ResolveFrozenActorOrder(this Actor self, Order order, Color
 			if (frozenLayer == null)
 				return Target.Invalid;
 
+			frozenId = order.ExtraData;
 			var frozen = frozenLayer.FromID(order.ExtraData);
 			if (frozen == null)
 				return Target.Invalid;
 
 			// Flashes the frozen proxy
 			self.SetTargetLine(frozen, targetLine, true);
 
-			// Target is still alive - resolve the real order
-			if (frozen.Actor != null && frozen.Actor.IsInWorld)
-				return Target.FromActor(frozen.Actor);
-
-			if (!order.Queued)
-				self.CancelActivity();
-
-			var move = self.TraitOrDefault<IMove>();
-			if (move != null)
-			{
-				// Move within sight range of the frozen actor
-				var sight = self.TraitOrDefault<RevealsShroud>();
-				var range = sight != null ? sight.Range : WDist.FromCells(2);
-
-				self.QueueActivity(move.MoveWithinRange(Target.FromPos(frozen.CenterPosition), range));
-			}
-
-			return Target.Invalid;
+			return Target.FromPos(frozen.CenterPosition);
```

Comment:
You can avoid all the `uint frozenId` argument changes by using the FrozenActor target type.
The simplest way to do that is to use `Target.FromFrozenActor` here, but you could in principle fold this logic into the `Target.FromOrder` above.

This had been the plan all along, but it wasn't implemented originally because the attack code of the time was a mess that couldn't easily be adapted to work with frozen actors.  We've rewritten most of the problematic parts since then, so you should just need to change your `frozenId` checks to `t.Type == TargetType.FrozenActor` now.